### PR TITLE
Deprecate position prop for PageLayout.Pane, update stories and tests.

### PR DESCRIPTION
**Pull request by radglob**
Deprecates the `position` prop for `PageLayout.Pane`. Partially addresses https://github.com/github/primer/issues/1199 in conjunction with https://github.com/primer/react/pull/3386.

### Screenshots

Please provide before/after screenshots for any visual changes

### Merge checklist

- [X] Added/updated tests
- [X] Added/updated documentation
- [X] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [X] Tested in Chrome
- [X] Tested in Firefox
- [X] Tested in Safari
- [X] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by mperrotti** [comment]
These changes look right to me. @colebemis - could you look at this one too?

File: src/PageLayout/PageLayout.tsx [hunk 1]
```diff
@@ -548,6 +560,13 @@ const Pane = React.forwardRef<HTMLDivElement, React.PropsWithChildren<PageLayout
     },
     forwardRef,
   ) => {
+    if (responsivePosition !== undefined) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'The `position` prop will be removed on the next major version. You should order your markup as you want it to render instead.',
+      )
+    }
+
```
Comment (colebemis) [inline src/PageLayout/PageLayout.tsx:569]: Love this incremental approach to breaking changes ❤️

File: src/PageLayout/PageLayout.stories.tsx [hunk 2]
```diff
@@ -205,31 +202,6 @@ const meta: Meta = {
       table: {category: 'Content props'},
     },
 
-    // Pane prop controls
```
Comment (colebemis) [inline src/PageLayout/PageLayout.stories.tsx:208]: We should also remove these controls from the SplitPageLayout stories.

File: src/PageLayout/PageLayout.tsx [hunk 3]
```diff
@@ -107,7 +107,20 @@ const Root: React.FC<React.PropsWithChildren<PageLayoutProps>> = ({
...
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            sx={(theme: any) => ({
+              display: 'flex',
+              flex: '1 1 100%',
+              flexWrap: 'wrap',
+              maxWidth: '100%',
+              [`@media screen and (min-width: ${theme.breakpoints[1]})`]: {
+                columnGap: SPACING_MAP[columnGap],
+              },
+            })}
```
Comment (colebemis) [inline src/PageLayout/PageLayout.tsx:120]: Looks like the gap between content and pane collapses on narrow screens. We probably need a `rowGap` in here somewhere.

No gap between content and pane regions

File: src/PageLayout/PageLayout.docs.json [hunk 4]
```diff
@@ -137,7 +137,8 @@
           "name": "position",
           "type": "| 'start' | 'end' | { narrow?: | 'start' | 'end' regular?: | 'start' | 'end' wide?: | 'start' | 'end' }",
           "defaultValue": "'end'",
-          "description": ""
+          "description": "",
+          "deprecated": true
```
Comment (colebemis) [inline src/PageLayout/PageLayout.docs.json:141]: We should also deprecate `position` in  `SplitPageLayout.docs.json`

**Review by colebemis — CHANGES_REQUESTED** [review]
Great work on this @radglob!

The only blocking feedback I have is that the gap between the content and pane regions disappears on narrow viewports. We should fix that before shipping this change.

I also made a couple comments about updating the stories and documentation of SplitPageLayout to match PageLayout.

**Pull request merged**